packages/lesswrong/lib/utils/viewUtils.ts

```diff
@@ -13,3 +22,114 @@ export function describeTerms(terms: ViewTermsBase) {
+  const logger = loggerConstructor(`views-${collectionName.toLowerCase()}`)
+  logger('getParameters(), terms:', terms);
+
+  let parameters: any = {
+    selector: {},
+    options: {},
+  };
+
+  if (collection.defaultView) {
+    parameters = merge(
+      parameters,
+      collection.defaultView(terms, apolloClient, context)
+    );
+    logger('getParameters(), parameters after defaultView:', parameters)
+  }
+
+  // handle view option
+  if (terms.view && collection.views[terms.view]) {
+    const viewFn = collection.views[terms.view];
+    const view = viewFn(terms, apolloClient, context);
+    let mergedParameters = merge(parameters, view);
+
+    if (
+      mergedParameters.options &&
+      mergedParameters.options.sort &&
+      view.options &&
+      view.options.sort
+    ) {
+      // If both the default view and the selected view have sort options,
+      // don't merge them together; take the selected view's sort. (Otherwise
+      // they merge in the wrong order, so that the default-view's sort takes
+      // precedence over the selected view's sort.)
+      mergedParameters.options.sort = view.options.sort;
+    }
+    parameters = mergedParameters;
+    logger('getParameters(), parameters after defaultView and view:', parameters)
+  }
+
+  // sort using terms.orderBy (overwrite defaultView's sort)
+  if (terms.orderBy && !_.isEmpty(terms.orderBy)) {
+    parameters.options.sort = terms.orderBy;
+  }
+
+  // if there is no sort, default to sorting by createdAt descending
+  if (!parameters.options.sort) {
+    parameters.options.sort = { createdAt: -1 } as any;
+  }
+
+  // extend sort to sort posts by _id to break ties, unless there's already an id sort
+  // NOTE: always do this last to avoid overriding another sort
+  if (!(parameters.options.sort && typeof parameters.options.sort._id !== undefined)) {
+    parameters = merge(parameters, { options: { sort: { _id: -1 } } });
+  }
+
+  // remove any null fields (setting a field to null means it should be deleted)
+  parameters.selector = replaceSpecialFieldSelectors(parameters.selector);
+  if (parameters.options.sort) {
+    _.keys(parameters.options.sort).forEach(key => {
+      if (parameters.options.sort[key] === null) {
+        delete parameters.options.sort[key];
+      }
+    });
+  }
+
+  // limit number of items to 1000 by default
+  const maxDocuments = maxDocumentsPerRequestSetting.get();
+  const limit = terms.limit || parameters.options.limit;
+  parameters.options.limit = !limit || limit < 1 || limit > maxDocuments ? maxDocuments : limit;
+
+  logger('getParameters(), final parameters:', parameters);
+  return parameters;
+}
+
+function replaceSpecialFieldSelectors(selector: any): any {
+  let result: any = {};
+  for (let key of Object.keys(selector)) {
+    if (_.isEqual(selector[key], viewFieldNullOrMissing)) {
+      // Put an explicit null in the selector. In mongodb-query terms, this means
+      // no constraint on the given field.
```

Comment:
> no constraint

This isn't correct to my reading. I read that comment as saying "'anything goes' for this field." However, the null is explicitly constraining the results to be either null or missing.

Reply:
Ah, yep, brain fart. (The `viewFieldNullOrMissing` identifier should've made it clear.)